internal/tests/api/scopes/scope_test.go

```diff
@@ -152,3 +155,102 @@ func TestErrors(t *testing.T) {
 	assert.NotNil(apiErr)
 	assert.EqualValues(http.StatusBadRequest, apiErr.Response().StatusCode())
 }
+
+func TestKeyDestruction(t *testing.T) {
+	ctx := context.Background()
+	tc := controller.NewTestController(t, &controller.TestControllerOpts{SchedulerRunJobInterval: time.Second})
+	t.Cleanup(tc.Shutdown)
+	c := tc.Client()
+	c.SetToken(tc.Token().Token)
+	sc := scopes.NewClient(c)
+
+	keys, err := sc.ListKeys(ctx, "global")
+	require.NoError(t, err)
+	assert.Len(t, keys.Items, 7)
+	for _, key := range keys.Items {
+		assert.Len(t, key.Versions, 1)
+	}
+
+	_, err = sc.RotateKeys(ctx, "global", false)
+	require.NoError(t, err)
+
+	keys, err = sc.ListKeys(ctx, "global")
+	require.NoError(t, err)
+	assert.Len(t, keys.Items, 7)
+	for _, key := range keys.Items {
+		assert.Len(t, key.Versions, 2)
+	}
+
+	// Root key is always last, by virtue of sorting by ID
+	rootKeyVersion := keys.Items[len(keys.Items)-1].Versions[1]
+	var destroyKeyVersion *scopes.KeyVersion
+	for _, key := range keys.Items {
+		if key.Purpose == kms.KeyPurposeDatabase.String() {
+			destroyKeyVersion = key.Versions[1]
+		}
+	}
+
+	jobs, err := sc.ListKeyVersionDestructionJobs(ctx, "global")
+	require.NoError(t, err)
+	assert.Len(t, jobs.Items, 0)
+
+	result, err := sc.DestroyKeyVersion(ctx, "global", rootKeyVersion.Id)
+	require.NoError(t, err)
+	assert.Equal(t, "completed", result.State)
+
+	jobs, err = sc.ListKeyVersionDestructionJobs(ctx, "global")
+	require.NoError(t, err)
+	assert.Len(t, jobs.Items, 0)
+
+	keys, err = sc.ListKeys(ctx, "global")
+	require.NoError(t, err)
+	assert.Len(t, keys.Items, 7)
+	for _, key := range keys.Items {
+		switch key.Purpose {
+		case "rootKey":
+			assert.Len(t, key.Versions, 1)
+		default:
+			assert.Len(t, key.Versions, 2)
+		}
+	}
+
+	result, err = sc.DestroyKeyVersion(ctx, "global", destroyKeyVersion.Id)
+	require.NoError(t, err)
+	assert.Equal(t, "pending", result.State)
+
+	jobs, err = sc.ListKeyVersionDestructionJobs(ctx, "global")
+	require.NoError(t, err)
+	assert.Len(t, jobs.Items, 1)
+
+	// The configured scheduler monitoring interval is 1 second. The jobs become available 1
+	// second after the last successful run. We need to re-encrypt data in 4 different tables,
+	// and then we need to destroy the key. This job will take between 4 and 5 seconds to run,
+	// depending on the timing of the first started run.
```

Comment:
1. only 4 tables?
2. is there any data to rewrap in tests? I didn't think that the test setup inserted any default data into the tables, and I didn't see you insert any in the test itself

Reply:
Yep! There's 8 pieces of encrypted data created by the test controller setup (creating some tokens, users, auth methods, etc) spread across 4 tables in the global scope.